python/iceberg/parquet/parquet_reader.py

```diff
@@ -0,0 +1,340 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from collections import OrderedDict
+from datetime import datetime
+import decimal
+import logging
+
+import fastparquet
```

Comment:
What is the reason for using `fastparquet` and `pyarrow` at the same time? `pyarrow` should provide all features you need.

Reply:
Let me review your code in kartothek because I did want to coalesce on just pyarrow.  I coded a lot of this a year ago, and I think the arrow project has moved quite a bit since then.  It's also possible some of what I was missing was there I just couldn't find it at the time without searching through the c code.